platform/mellanox/mlnx-platform-api/sonic_platform/psu.py

```diff
@@ -122,6 +130,24 @@ def get_name(self):
         return self._name
 
 
+    def _read_vpd_file(self, filename):
+        """
+        Read a vpd file parsed from eeprom with keys and values.
+        Returns a dictionary.
+        """
+        result = {}
+        try:
+            if not os.path.exists(filename):
+                return result
+            with open(filename, 'r') as fileobj:
+                for line in fileobj.readlines():
```

Comment:
Do we want to do some checks to ensure both keys are available here?

Reply:
This is intended to be a generic function for reading this vpd style file. I wouldn't want to specialize it. It will simply return an empty dictionary if there are no keys present in the file (or the file does not exist). This case can be handled in the specific logic parsing the output of this function. 

See @Junchao-Mellanox comments for the open discussion on if we should simply acknowledge in the log as an info that the data is not present or whether an error should be thrown in the log. Either way, my opinion is that we should handle any errors gracefully and simply log the problem so that the Platform API can continue to work.

Reply:
It seems we should have all of this information available on all platforms except 2010 and 2100 so I can hardcode those two platforms in and send an error to the log if the FRU data is not available on the other platforms. This is a design decision I don't have a strong opinion on because I am unfamiliar with the general best practice in SONiC specifically for this situation.

Reply:
agree that this function shall be generic. I would suggest in the below code check the key existence first(if it's expected to be available like none 2010/2100 platform), then we can print a more user-friendly log to indicate that some field does not exist, instead of a python key error msg, what do you think? 
```
        self.model = self._read_vpd_file(self.psu_vpd).get(PN_VPD_FIELD, "")
        self.serial = self._read_vpd_file(self.psu_vpd).get(SN_VPD_FIELD, "")
```

Reply:
Sounds good to me, will implement and submit for review.